nixos/modules/services/networking/nordvpn.nix

```diff
@@ -0,0 +1,107 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.nordvpn;
+  package = pkgs.nordvpn;
+in {
+  options.services.nordvpn = {
+    enable = mkEnableOption "nordvpn";
+
+    connect = {
+      technology = mkOption {
+        type = types.str;
+        default = "OpenVPN";
+        description = ''
+          Technology to use in NordVPN , two possible option are 'OpenVPN' and 'NordLynx'
+        '';
+      };
+
+      autoconnect = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Autoconnect to NordVPN using NixOS options
+        '';
+      };
+
+      username = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          Username to use during NordVPN login
+        '';
+      };
+
+      passwordFile = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          Password file to use during NordVPN login
+        '';
+      };
+
+      connectLocation = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          Connection location for NordVPN , it can be country code, location, etc
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    boot.kernelModules = [ "tun" ];
+    networking.iproute2.enable = true;
+
+    users.groups.nordvpn = {};
+
+    systemd.packages = [ package ];
+
+    systemd.sockets.nordvpnd.wantedBy = [ "sockets.target" ];
+
+    systemd.services.nordvpnd = {
+      wantedBy = ["default.target"];
+      path = with pkgs; [
+        iptables
+        iproute2
+        sysctl
+      ];
+
+      # NOTE: the preStart process is needed to provide RSA key and server list for the nordvpn daemon
+      preStart = ''
+        ${pkgs.xorg.lndir}/bin/lndir ${cfg.package}/var/lib/nordvpn /var/lib/nordvpn
+      '';
+
+      serviceConfig = {
+        ExecStart = "${package}/bin/nordvpnd";
+        StateDirectory = "nordvpn";
+      };
+    };
+
+    systemd.services.nordvpnd-login = mkIf cfg.connect.autoconnect {
+      wantedBy = [ "default.target" ];
+      after = [ "nordvpnd.service" ];
+
+      path = [
+        cfg.package
+      ];
+
+      script = ''
+        nordvpn disconnect
+        nordvpn set technology ${cfg.connect.technology}
+        nordvpn logout
+        nordvpn login --username ${cfg.connect.username} --password "$(cat ${cfg.connect.passwordFile})"
```

Comment:
Would it be possible to use `pass` instead of `cat` with bring-your-own-password-echoer? [`himalaya`](https://github.com/soywod/himalaya) does this by default where you must provide a command that will called to get the password. If not I should at least be able to use process substitution like `fish`'s `psub` or `bash`'s `<(process)`? Really what I'm trying to understand is how to store this password not in plaintext.

Reply:
Is there existing nixos module that utilize similar approach?

Reply:
I am not the person to ask about that :sweat_smile: